### PR TITLE
Separated out engine compilation from game files

### PR DESCRIPTION
**Pull request by J-Mo63**

### CODE REVIEW

**Comment by J-Mo63** [comment]
After doing some further research into the implementation and benefits of using dynamic linking for the engine, I have come to the conclusion that we should continue to use static libs for both ease of use and efficiency reasons. Any release version of a game using the engine should be statically compiled anyway, so may as well do it in all cases.

**Comment by J-Mo63** [comment]
Despite the failing Windows tests, I would like to submit the PR for merge as [the issue seems to be entirely related to raylib](https://github.com/raysan5/raylib/issues/1153), which will hopefully be phased out over the coming months anyway (staring with this PR).

File: vendor/engine/src/utils/Maths.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+bool BoundedBox::Intersects(const BoundedBox &other) const
+{
+    if ((max.x >= other.min.x) && (min.x <= other.max.x))
+    {
+        if ((max.y < other.min.y) || (min.y > other.max.y)) return false;
+        if ((max.z < other.min.z) || (min.z > other.max.z)) return false;
+    }
+    else return false;
+    return true;
+}
```
Comment (Raelr) [inline vendor/engine/src/utils/Maths.cpp:32]: Note: we should probably pull this out into a Vector class

File: vendor/engine/src/utils/Maths.h [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+#define VEC_OPERATOR_BODY_VEC(op, sep) \
+    x op rhs.x sep \
+    y op rhs.y sep \
+    z op rhs.z sep
+
+#define VEC_OPERATOR_BODY_FLOAT(op, sep) \
+    x op rhs sep \
+    y op rhs sep \
+    z op rhs sep
```
Comment (Raelr) [inline vendor/engine/src/utils/Maths.h:33]: Either commment this or remove it for the sake of readability

**Review by Raelr — APPROVED** [review]
Disgusting. Never seen worse code in my ENTIRE life. I demand to see the manager. LGTM 👍

**Pull request merged**